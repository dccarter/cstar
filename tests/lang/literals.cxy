/*
 * Tests literals in cxy
 */

func main(args: [string]) {
    // null literal
    null;
    // bool literal
    true;
    false;
    // char literals
    'a'; 'b'; '\n'; '\0'; 'ç•Œ'; 'ðŸ˜Š';
    // integers
    0; 100; 18446744073709551615;
    -1; -2; -9223372036854775807;
    0b100; 0b1010;
    0o777; 0o189;
    0xa; 0xff; 0xffffff;
    // floats
    1.100; 1.000; 10.0e5; 1.9e-5;
    // strings
    ""; "Hello World"; "Hello ä¸–ç•Œ";
}
