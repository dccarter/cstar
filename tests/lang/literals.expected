{
 "tag": "Program",
 "decls": [{
   "tag": "FuncDecl",
   "flags": "(TopLevelDecl)",
   "overload": 51,
   "index": 0,
   "name": "main",
   "params": [{
     "tag": "FuncParam",
     "name": "args",
     "index": 0,
     "paramType": {
      "tag": "ArrayType",
      "flags": "(TypeAst)",
      "element": {
       "tag": "StringType",
       "flags": "(TypeAst)"
      }
     }
    }],
   "body": {
    "tag": "BlockStmt",
    "stmts": [{
      "tag": "ExprStmt",
      "expr": {
       "tag": "NullLit",
       "value": null
      }
     }, {
      "tag": "ExprStmt",
      "expr": {
       "tag": "BoolLit",
       "value": 1
      }
     }, {
      "tag": "ExprStmt",
      "expr": {
       "tag": "BoolLit",
       "value": 0
      }
     }, {
      "tag": "ExprStmt",
      "expr": {
       "tag": "CharLit",
       "value": 97
      }
     }, {
      "tag": "ExprStmt",
      "expr": {
       "tag": "CharLit",
       "value": 98
      }
     }, {
      "tag": "ExprStmt",
      "expr": {
       "tag": "CharLit",
       "value": 10
      }
     }, {
      "tag": "ExprStmt",
      "expr": {
       "tag": "CharLit",
       "value": 0
      }
     }, {
      "tag": "ExprStmt",
      "expr": {
       "tag": "CharLit",
       "value": 30028
      }
     }, {
      "tag": "ExprStmt",
      "expr": {
       "tag": "CharLit",
       "value": 128522
      }
     }, {
      "tag": "ExprStmt",
      "expr": {
       "tag": "IntegerLit",
       "value": 0
      }
     }, {
      "tag": "ExprStmt",
      "expr": {
       "tag": "IntegerLit",
       "value": 100
      }
     }, {
      "tag": "ExprStmt",
      "expr": {
       "tag": "IntegerLit",
       "value": -1
      }
     }, {
      "tag": "ExprStmt",
      "expr": {
       "tag": "UnaryExpr",
       "op": 24,
       "isPrefix": true,
       "lhs": {
        "tag": "IntegerLit",
        "value": 1
       }
      }
     }, {
      "tag": "ExprStmt",
      "expr": {
       "tag": "UnaryExpr",
       "op": 24,
       "isPrefix": true,
       "lhs": {
        "tag": "IntegerLit",
        "value": 2
       }
      }
     }, {
      "tag": "ExprStmt",
      "expr": {
       "tag": "UnaryExpr",
       "op": 24,
       "isPrefix": true,
       "lhs": {
        "tag": "IntegerLit",
        "value": 9223372036854775808
       }
      }
     }, {
      "tag": "ExprStmt",
      "expr": {
       "tag": "IntegerLit",
       "value": 4
      }
     }, {
      "tag": "ExprStmt",
      "expr": {
       "tag": "IntegerLit",
       "value": 10
      }
     }, {
      "tag": "ExprStmt",
      "expr": {
       "tag": "IntegerLit",
       "value": 511
      }
     }, {
      "tag": "ExprStmt",
      "expr": {
       "tag": "IntegerLit",
       "value": 1
      }
     }, {
      "tag": "ExprStmt",
      "expr": {
       "tag": "IntegerLit",
       "value": 89
      }
     }, {
      "tag": "ExprStmt",
      "expr": {
       "tag": "IntegerLit",
       "value": 10
      }
     }, {
      "tag": "ExprStmt",
      "expr": {
       "tag": "IntegerLit",
       "value": 255
      }
     }, {
      "tag": "ExprStmt",
      "expr": {
       "tag": "IntegerLit",
       "value": 16777215
      }
     }, {
      "tag": "ExprStmt",
      "expr": {
       "tag": "FloatLit",
       "value": 1.100000
      }
     }, {
      "tag": "ExprStmt",
      "expr": {
       "tag": "FloatLit",
       "value": 1
      }
     }, {
      "tag": "ExprStmt",
      "expr": {
       "tag": "FloatLit",
       "value": 1000000
      }
     }, {
      "tag": "ExprStmt",
      "expr": {
       "tag": "FloatLit",
       "value": 0.000019
      }
     }, {
      "tag": "ExprStmt",
      "expr": {
       "tag": "StringLit",
       "value": ""
      }
     }, {
      "tag": "ExprStmt",
      "expr": {
       "tag": "StringLit",
       "value": "Hello World"
      }
     }, {
      "tag": "ExprStmt",
      "expr": {
       "tag": "StringLit",
       "value": "Hello 世界"
      }
     }]
   }
  }]
}
